source/RialtoGStreamerMSEBaseSink.cpp

```diff
@@ -1019,24 +1024,127 @@ void rialto_mse_base_sink_lost_state(RialtoMSEBaseSink *sink)
 
         if (g_object_class_find_property(G_OBJECT_GET_CLASS(parentObject), "flags"))
         {
             guint flags = 0;
             g_object_get(parentObject, "flags", &flags, nullptr);
             n_video = flags & rialto_mse_base_sink_get_gst_play_flag("video") ? n_video : 0;
             n_audio = flags & rialto_mse_base_sink_get_gst_play_flag("audio") ? n_audio : 0;
+            n_text = flags & rialto_mse_base_sink_get_gst_play_flag("text") ? n_text : 0;
         }
 
         return true;
     }
 
     return false;
 }
+
+static bool rialto_mse_base_sink_set_streams_number(RialtoMSEBaseSink *sink, GstObject *parentObject)
+{
+    RialtoMSEBaseSinkPrivate *priv = sink->priv;
+    int32_t videoStreams{-1}, audioStreams{-1}, subtitleStreams{-1};
+
+    GstContext *context = gst_element_get_context(GST_ELEMENT(sink), "streams-info");
+    if (context)
+    {
+        GST_DEBUG_OBJECT(sink, "Getting number of streams from \"streams-info\" context");
+
+        guint n_video{0}, n_audio{0}, n_text{0};
+
+        const GstStructure *streamsInfoStructure = gst_context_get_structure(context);
+        gst_structure_get_uint(streamsInfoStructure, "video-streams", &n_video);
+        gst_structure_get_uint(streamsInfoStructure, "audio-streams", &n_audio);
+        gst_structure_get_uint(streamsInfoStructure, "text-streams", &n_text);
+
+        if (n_video > std::numeric_limits<int32_t>::max() || n_audio > std::numeric_limits<int32_t>::max() ||
+            n_text > std::numeric_limits<int32_t>::max())
+        {
+            GST_ERROR_OBJECT(sink, "Number of streams is too big, video=%u, audio=%u, text=%u", n_video, n_audio, n_text);
+            gst_context_unref(context);
+            return false;
+        }
+
+        videoStreams = n_video;
+        audioStreams = n_audio;
+        subtitleStreams = n_text;
+
+        gst_context_unref(context);
+    }
+    else if (rialto_mse_base_sink_get_n_streams_from_parent(parentObject, videoStreams, audioStreams, subtitleStreams))
+    {
+        GST_DEBUG_OBJECT(sink, "Got number of streams from playbin2 properties");
+    }
+    else
+    {
+        std::lock_guard<std::mutex> lock(priv->m_sinkMutex);
+        if (priv->m_mediaSourceType == firebolt::rialto::MediaSourceType::VIDEO)
+        {
+            videoStreams = priv->m_numOfStreams;
+            if (priv->m_isSinglePathStream)
+            {
+                audioStreams = 0;
+                subtitleStreams = 0;
+            }
+        }
+        else if (priv->m_mediaSourceType == firebolt::rialto::MediaSourceType::AUDIO)
+        {
+            audioStreams = priv->m_numOfStreams;
+            if (priv->m_isSinglePathStream)
+            {
+                videoStreams = 0;
+                subtitleStreams = 0;
+            }
+        }
+        else if (priv->m_mediaSourceType == firebolt::rialto::MediaSourceType::SUBTITLE)
+        {
+            subtitleStreams = priv->m_numOfStreams;
+            if (priv->m_isSinglePathStream)
```

Comment:
Does the isSinglePathStream variable still work now we have 3 streams? Could it be possible to have audio + subtitle & no video?